base_global_discount/models/global_discount.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+from odoo.addons import decimal_precision as dp
+
+
+class GlobalDiscount(models.Model):
+    _name = 'global.discount'
+    _description = 'Global Discount'
+    _order = "sequence, id desc"
+
+    sequence = fields.Integer(
+        help='Gives the order to apply discounts',
+    )
+    name = fields.Char(
+        string='Discount Name',
+        required=True,
+    )
+    discount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        required=True,
+        default=0.0,
+    )
+    discount_scope = fields.Selection(
+        selection=[
+            ('sale', 'Sales'),
+            ('purchase', 'Purchases'),
+        ],
+        default='sale',
+        required='True',
+        string='Discount Scope',
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        default=lambda self: self.env.user.company_id,
+    )
+
+    def name_get(self):
+        result = []
+        for one in self:
+            result.append(
+                (one.id, '{} ({:.2f}%)'.format(one.name, one.discount)))
```

Comment:
I think that here can avoid for loop usage:
```
self.mapped(lambda x: (x.id, '{} ({:.2f}%)'.format(x.name, x.discount)))
```

Reply:
Doesn't it do it below the surface anyway?

Reply:
Yes, but use mapped it's more "Odoonic"... if tomorrow odoo improve their code you can get the benefits without modify your code. And it's more readable no?

What do you think?

Reply:
I can buy you the first argument (anyway this model is meant to keep just a handful of registers), but the same could be speculated with the current form.

In the readable part, I'd say the oneliners tend to be more obscure.

Reply:
I see it as syntactic sugar, but of course it's an opinion... totally agree if you don't want use it here.